server/middleware/auth.js

```diff
@@ -0,0 +1,26 @@
+const express = require("express");
+const jwt = require("jsonwebtoken");
+const { User } = require("../models/user.models");
+
+const auth = async (req, res, next) => {
+  try {
+    if (!req.cookies.authenticationToken) {
+      return res.status(401).send({ responseMessage: "Authentication error!" });
+    }
+
+    const decodedToken = jwt.verify(
+      req.cookies.authenticationToken,
+      process.env.SECRET_KEY
+    );
+    const userId = decodedToken.userId;
+    const user = await User.findOne({ _id: userId });
+    //Allows access of user object in authenticated routes by accessing req.user
+    req.user = user;
+
+    next();
```

Comment:
yeah this is fine. it all depends on how we want to use our app but now we can just work with this